src/components/panel/panel.js

```diff
@@ -1282,18 +1288,24 @@ MdPanelRef.prototype._animateOpen = function() {
  * @private
  */
 MdPanelRef.prototype._animateClose = function() {
+  var onRemoving = this._config['onRemoving'];
+
   var animationConfig = this._config['animation'];
   if (!animationConfig) {
     this.removeClass('md-panel-is-showing');
     this.removeClass('_md-panel-shown');
+    this._$q.when(this._config['scope'].$eval(onRemoving))
     return this._$q.resolve();
   }
 
   var self = this;
   return this._$q(function(resolve, reject) {
-    animationConfig.animateClose(self._$q).then(function(){
+    animationConfig.animateClose(self._$q).then(function() {
       self.removeClass('md-panel-is-showing');
-      resolve(self);
+      onRemoving = self._$q.when(this._config['scope'].$eval(onRemoving))
```

Comment:
The 'this' would be undefined at this point. Use 'self'. Also please verify (in a test or demo) that the callbacks are in the proper scope (using the caller's 'this' value).